File: Artisphere/src/main/java/com/skilldistillery/museums/services/MuseumService.java
package com.skilldistillery.museums.services;

import java.util.List;
import com.skilldistillery.artisphere.entities.Museum;

public interface MuseumService {
    Museum saveMuseum(Museum museum);
    Museum getMuseumById(int id);
    List<Museum> getAllMuseums();
    boolean deleteMuseum(int id);
	Museum create(Museum newMuseum);
}

File: museumApp/src/app/services/museum.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Museum } from '../models/museum';

@Injectable({
  providedIn: 'root'
})
export class MuseumService {

  private baseUrl = 'http://localhost:8085/api/museums';  // Adjust this URL based on your backend

  constructor(private http: HttpClient) { }

  // Get all museums
  getAllMuseums(): Observable<Museum[]> {
    return this.http.get<Museum[]>(this.baseUrl);
  }

  // Get museum by ID
  getMuseumById(id: number): Observable<Museum> {
    return this.http.get<Museum>(`${this.baseUrl}/${id}`);
  }

  // Create a new museum
  createMuseum(museum: Museum): Observable<Museum> {
    return this.http.post<Museum>(this.baseUrl, museum);
  }
}
